judo-runtime-core-jsl-itest/models/EntityRelationshipsModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/CompositionRelationshipsTest.java

```diff
@@ -568,4 +557,139 @@ void testDeepCopyUpdate() {
                 () -> entityADao.update(a5)
         );
     }
+
+    @Test
+    void testAddAndRemoveOnCollections() {
+        EntityC c2 = entityCDao.create(EntityC.builder().withStringC("C2").build());
+        EntityC c3 = entityCDao.create(EntityC.builder().withStringC("C3").build());
+        EntityA entityA1 = entityADao.create(EntityA.builder()
+                .withSingleRequiredConA(EntityC.builder().withStringC("C1").build())
+                .withCollectionConA(List.of(c2, c3))
+                .build());
+
+        assertEquals("C1", entityA1.getSingleRequiredConA().getStringC().orElseThrow());
+        assertEquals(2, entityA1.getCollectionConA().size());
+        assertEquals("C2", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C2")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C3", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C3")).findFirst().orElseThrow().getStringC().orElseThrow());
+
+        EntityC c4 = EntityC.builder().withStringC("C4").build();
+        EntityC c5 = EntityC.builder().withStringC("C5").build();
+        EntityC c6 = EntityC.builder().withStringC("C6").build();
+        entityA1.addToCollectionConA(c4);
+        entityA1.addToCollectionConA(c5, c6);
+
+        assertEquals("C1", entityA1.getSingleRequiredConA().getStringC().orElseThrow());
+        assertEquals(5, entityA1.getCollectionConA().size());
+        assertEquals("C2", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C2")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C3", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C3")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C4", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C4")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C5", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C5")).findFirst().orElseThrow().getStringC().orElseThrow());
+        assertEquals("C6", entityA1.getCollectionConA().stream().filter(c -> c.getStringC().orElseThrow().equals("C6")).findFirst().orElseThrow().getStringC().orElseThrow());
+
+        entityA1.addToCollectionConA(null);
+        assertEquals(6, entityA1.getCollectionConA().size());
+        EntityC c7 = EntityC.builder().withStringC("C7").build();
+        entityA1.addToCollectionConA(null, c7);
```

Comment:
I believe null values should not be permitted as they have the potential to obscure bugs in the event that variables unintentionally become null.

Reply:
Currently, null values can also be specified in the set call. The reason I allowed it to add a null value in the add method was to make set and add consistent with each other.